docs: Add vm-delete-protection operand (#1216)
kubevirt/ssp-operator: docs/development.md

metrics can be found in `docs/metrics.md`. The installed rules can be found in
`internal/operands/metrics/resources.go`.

#### `vm-delete-protection`

Installs a [ValidatingAdmissionPolicy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)

> (VAP) and ValidatingAdmissionPolicyBinding which bounds the VAP cluster-wide. They prevent VirtualMachine (VM) objects

Reviewer: ```suggestion
(VAP) and ValidatingAdmissionPolicyBinding which binds the VAP to all namespaces. The VAP prevents VirtualMachine (VM) objects
```

Author: Done.

> (VAP) and ValidatingAdmissionPolicyBinding which binds the VAP to all namespaces. The VAP prevents VirtualMachine (VM) objects

Reviewer 2: `ValidatingAdmissionPolicyBinding (VAP Binding)` is missing a link to docs, maybe link it to [1]. 

[1] https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy

Author: Done.

from being deleted when protection is enabled. To enable the delete protection, the label

> `kubevirt.io/vm-delete-protection: "True"` has to be added to the VM object. The installed VAP can

Reviewer: ```suggestion
`kubevirt.io/vm-delete-protection: "True"` has to be added to the VM object. The generated resources can
```

Author: Done.

metrics can be found in `docs/metrics.md`. The installed rules can be found in
`internal/operands/metrics/resources.go`.

#### `vm-delete-protection`

Installs a [ValidatingAdmissionPolicy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)
(VAP)
and [ValidatingAdmissionPolicyBinding](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy)
which binds the VAP to all namespaces. The VAP prevents VirtualMachine (VM) objects
from being deleted when protection is enabled. To enable the delete protection, the label
`kubevirt.io/vm-delete-protection: "True"` has to be added to the VM object. The generated resources VAP can

> be found in `internal/operands/vm-delete-protection/resources.go`.

Reviewer 3: Can you change the word order to:
```
The generated VAP resources...
```
It will be easier to read.
